mne/io/meas_info.py

```diff
@@ -157,7 +248,83 @@ def copy(self):
         info : instance of Info
             The copied info.
         """
-        return Info(super(Info, self).copy())
+        return Info(self._store)
+
+    def to_dict(self):
+        """Obtain a version of this info object in the form of a plain Python
+        dict
+
+        While the Info object behaves like a dict, it is a subclass of
+        MutableMapping. Use this function to cast it to an actual dict. Be
+        aware that the read-only fields will become writable and will no longer
+        auto-update.
+
+        Notes
+        -----
+        - To cast the plain dict back to an instance of Info, use:
+          `mne.io.Info(info_dict)`
+        - Always use a propert instance of Info when calling MNE-Python
+          functions.
+        """
+        info_dict = dict(self)
+        # Convert non-standard fields
+        info_dict['ch_names'] = list(info_dict['ch_names'])
+        return info_dict
+
+    def __getitem__(self, key):
+        """Retrieve a value from the data store
+
+        Parameters
+        ----------
+        key : str
+            The key associated with the value to retrieve
+
+        Returns
+        -------
+        value : object
+            The value associated with the key
+        """
+        try:
+            return Info._read_only_fields[key](self)
+        except KeyError:
+            return self._store[key]
+
+    def __setitem__(self, key, value):
+        """Store a value in the data store
+
+        Parameters
+        ----------
+        key : str
+            The key associated with the value for later retrieval
+        value : object
+            The value associated with the key
+        """
+        if key in Info._read_only_fields:
+            raise ValueError('The field ' + key + ' is read only.')
```

Comment:
'The field ' + key + ' is read only.'
->
'The field %s is read only.' % key

otherwise it will fail if key is an int for example